packages/backend/src/outgoing_payment/service.test.ts

```diff
@@ -0,0 +1,771 @@
+    jest.restoreAllMocks()
+
+    for (const plugin of Object.values(plugins)) {
+      // Plugins must be cleaned up, otherwise ilp-plugin-http can leak http2 connections.
+      expect(plugin.isConnected()).toBe(false)
+    }
+    plugins = {}
+  })
+
+  afterAll(
+    async (): Promise<void> => {
+      await appContainer.shutdown()
+      await truncateTables(knex)
+    }
+  )
+
+  describe('get', (): void => {
+    it('returns undefined when no payment exists', async () => {
+      await expect(outgoingPaymentService.get(uuid())).resolves.toBeUndefined()
+    })
+  })
+
+  describe('create', (): void => {
+    it('creates an OutgoingPayment', async () => {
+      const payment = await outgoingPaymentService.create({
+        superAccountId,
+        paymentPointer: 'http://wallet.example/paymentpointer/bob',
+        amountToSend: BigInt(123),
+        autoApprove: false
+      })
+      expect(payment.state).toEqual(PaymentState.Inactive)
+      expect(payment.intent).toEqual({
+        paymentPointer: 'http://wallet.example/paymentpointer/bob',
+        amountToSend: BigInt(123),
+        autoApprove: false
+      })
+      expect(payment.superAccountId).toBe(superAccountId)
+      await expect(
+        connectorService.getAccountBalance(payment.sourceAccount.id)
+      ).resolves.toEqual({ balance: BigInt(0) })
+      expect(payment.sourceAccount.code).toBe('USD')
+      expect(payment.sourceAccount.scale).toBe(9)
+      expect(payment.destinationAccount).toEqual({
+        scale: 9,
+        code: 'XRP',
+        url: 'http://wallet.example/paymentpointer/bob'
+      })
+
+      const payment2 = await outgoingPaymentService.get(payment.id)
+      if (!payment2) throw 'no payment'
+      expect(payment2.id).toEqual(payment.id)
+    })
+  })
+
+  describe('processNext', (): void => {
+    describe('Inactive→', (): void => {
+      it('Ready (FixedSend)', async (): Promise<void> => {
+        const paymentId = (
+          await outgoingPaymentService.create({
+            superAccountId,
+            paymentPointer: 'http://wallet.example/paymentpointer/bob',
+            amountToSend: BigInt(123),
+            autoApprove: false
+          })
+        ).id
+        const payment = await processNext(paymentId, PaymentState.Ready)
+        if (!payment.quote) throw 'no quote'
+        expect(payment.quote.timestamp).toBeInstanceOf(Date)
+        expect(
+          payment.quote.activationDeadline.getTime() - Date.now()
+        ).toBeGreaterThan(0)
+        expect(payment.quote.targetType).toBe(Pay.PaymentType.FixedSend)
+        expect(payment.quote.minDeliveryAmount).toBe(
+          BigInt(Math.ceil(123 * payment.quote.minExchangeRate.valueOf()))
+        )
+        expect(payment.quote.maxSourceAmount).toBe(BigInt(123))
+        expect(payment.quote.maxPacketAmount).toBe(
+          BigInt('9223372036854775807')
+        )
+        expect(payment.quote.minExchangeRate.valueOf()).toBe(
+          0.5 * (1 - Config.slippage)
```

Comment:
nit: get the config from `deps` incase someone alters or mocks the returned value